### ISSUE TITLE
Add L4 protocol label to advanced metrics

### ISSUE THREAD

**Issue by andreev-io** [microsoft/retina#239]
**Is your feature request related to a problem? Please describe.**
Advanced metrics include packet information such as IP and port. A typical packet 5-tuple is `(protocol, src IP, src port, dst IP, dst port)` (this would be a 3-tuple in the `Local Context` mode). This issue is to add the protocol label to such plugins as `dropreason` and `packetparser`.

**Describe the solution you'd like**
The protocol label should be optionally included in such metrics as `drop_count`, `drop_bytes`, `adv_forward_bytes`, `adv_forward_count`.

**Describe alternatives you've considered**
N/A

**Additional context**
N/A

**Comment by vakalapa** [comment]
Yeah the idea was to use additional labels field for optional labels we might need later on. But for this protocol field, current approach makes perfect sense.
